tests/integration/security/mtls/strict_test.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mtls
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/tests/integration/security/util/reachability"
+)
+
+// This test verifies reachability under different authN scenario when automtls enabled
+// - app A to app B using mTLS.
+// In each test, the steps are:
+// - Configure authn policy.
+// - Wait for config propagation.
+// - Send HTTP/gRPC requests between apps.
+func TestMtlsStrict(t *testing.T) {
+	framework.NewTest(t).
+		Run(func(ctx framework.TestContext) {
+
+			rctx := reachability.CreateContext(ctx, g, p)
+			systemNM := namespace.ClaimSystemNamespaceOrFail(ctx, ctx)
+
+			testCases := []reachability.TestCase{
+				{
+					ConfigFile:          "global-mtls-on-no-dr.yaml",
+					Namespace:           systemNM,
+					RequiredEnvironment: environment.Kube,
+					Include: func(src echo.Instance, opts echo.CallOptions) bool {
+						// Exclude calls to the headless TCP port.
+						if opts.Target == rctx.Headless && opts.PortName == "tcp" {
+							return false
+						}
+
+						return true
+					},
+					ExpectSuccess: func(src echo.Instance, opts echo.CallOptions) bool {
+						// When mTLS is in STRICT mode, DR's TLS settings are default to mTLS so the result would
+						// be the same as having global DR rule.
+						if opts.Target == rctx.Naked {
+							// calls to naked should always succeed.
+							return true
+						}
+
+						// If one of the two endpoints is naked, expect failure.
+						return src != rctx.Naked && opts.Target != rctx.Naked
```

Comment:
if source has envoy sidecar, dest does not, and we have autoMTLS turned out, it should succeed, right? do I miss anything? What's the transport_socket_match/tls_context config in that case?

Reply:
how about 
```
// If source is naked, and destination is not, expect failure.
return !(src == rctx.Naked && opts.Target != rctx.Naked)
```

Reply:
Yeah.. infact, i would do this as two tests @GregHanson [in a different PR]. one where mtls is enabled and one where it is disabled, but auto mtls is turned on in both. Secondly, even in the existing test, I think you have to remove the destination rule that we are adding in the helm charts, in order to truly test out the auto mTLS feature. From that point of view, it may actually be better to set global.mtls.enabled=false and global.mtls.auto=true so that we test the case where its all permissive [most common setup] and there are no dest rules.

Reply:
> if source is naked destination is not, expect failure under STRICT

This sounds right. But then I'll expect you to change the implementation, to make CDS also have `transport_socket_matches` configured as same as PERMISSIVE.

Yeah, otherwise, autoMTLS works great during PERMISSIVE, but if customers change to STRICT they'll need to configure DR again... 

I dont know the what DR this test is using as part of the Helm chart... If you want to make test hermetic, another option is to configure a dumb DR with empty TLS settings.